### PR TITLE
chore(GridForm): update jest + refactor tests

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Refactored GridForm test and updated Jest. 

This is just refactoring/deduping GridForm-related tests and bumping Jest to its newest version.  Fix for the bug ticket will be shipped in a follow up PR

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: web-1809
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

### Testing Instructions
Tests should pass ✨ 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/30716)  | [Monolith Env](https://pr-30716-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/1952)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-1952)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.122-alpha.454c5d.0<br/>@codecademy/gamut-labs@31.11.11-alpha.454c5d.0<br/>@codecademy/gamut@47.1.2-alpha.454c5d.0<br/>@codecademy/monaco@0.6.41-alpha.454c5d.0<br/>@codecademy/styleguide@57.0.11-alpha.454c5d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627942e9c8dd550fece0059e--gamut-preview.netlify.app>https://627942e9c8dd550fece0059e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627942e9c8dd550fece0059e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.122-alpha.c324d1.0<br/>@codecademy/gamut-labs@31.11.11-alpha.c324d1.0<br/>@codecademy/gamut@47.1.2-alpha.c324d1.0<br/>@codecademy/monaco@0.6.41-alpha.c324d1.0<br/>@codecademy/styleguide@57.0.11-alpha.c324d1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627c015d155c6414dc3cdd21--gamut-preview.netlify.app>https://627c015d155c6414dc3cdd21--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627c015d155c6414dc3cdd21>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.d3e2e6.0<br/>@codecademy/gamut-kit@0.6.122-alpha.d3e2e6.0<br/>@codecademy/gamut-labs@31.11.11-alpha.d3e2e6.0<br/>@codecademy/gamut@47.1.2-alpha.d3e2e6.0<br/>@codecademy/monaco@0.6.41-alpha.d3e2e6.0<br/>@codecademy/styleguide@57.0.11-alpha.d3e2e6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627c2a44a6497b0a7d13517c--gamut-preview.netlify.app>https://627c2a44a6497b0a7d13517c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627c2a44a6497b0a7d13517c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.8016aa.0<br/>@codecademy/gamut-kit@0.6.122-alpha.8016aa.0<br/>@codecademy/gamut-labs@31.11.11-alpha.8016aa.0<br/>@codecademy/gamut@47.1.2-alpha.8016aa.0<br/>@codecademy/monaco@0.6.41-alpha.8016aa.0<br/>@codecademy/styleguide@57.0.11-alpha.8016aa.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627c358bd41bb413192574fb--gamut-preview.netlify.app>https://627c358bd41bb413192574fb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627c358bd41bb413192574fb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.38d399.0<br/>@codecademy/gamut-kit@0.6.122-alpha.38d399.0<br/>@codecademy/gamut-labs@31.11.11-alpha.38d399.0<br/>@codecademy/gamut@47.1.2-alpha.38d399.0<br/>@codecademy/monaco@0.6.41-alpha.38d399.0<br/>@codecademy/styleguide@57.0.11-alpha.38d399.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.d1ad98.0<br/>@codecademy/gamut-kit@0.6.122-alpha.d1ad98.0<br/>@codecademy/gamut-labs@31.11.11-alpha.d1ad98.0<br/>@codecademy/gamut@47.1.2-alpha.d1ad98.0<br/>@codecademy/monaco@0.6.41-alpha.d1ad98.0<br/>@codecademy/styleguide@57.0.11-alpha.d1ad98.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d4fc842f2ce0dbcd93f74--gamut-preview.netlify.app>https://627d4fc842f2ce0dbcd93f74--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d4fc842f2ce0dbcd93f74>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.1a81e2.0<br/>@codecademy/gamut-kit@0.6.122-alpha.1a81e2.0<br/>@codecademy/gamut-labs@31.11.11-alpha.1a81e2.0<br/>@codecademy/gamut@47.1.2-alpha.1a81e2.0<br/>@codecademy/monaco@0.6.41-alpha.1a81e2.0<br/>@codecademy/styleguide@57.0.11-alpha.1a81e2.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.824015.0<br/>@codecademy/gamut-kit@0.6.125-alpha.824015.0<br/>@codecademy/gamut-labs@31.11.14-alpha.824015.0<br/>@codecademy/gamut@47.3.1-alpha.824015.0<br/>@codecademy/monaco@0.6.44-alpha.824015.0<br/>@codecademy/styleguide@57.2.1-alpha.824015.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d55df889f8f0cdd3bb41b--gamut-preview.netlify.app>https://627d55df889f8f0cdd3bb41b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d55df889f8f0cdd3bb41b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.d0d38c.0<br/>@codecademy/gamut-kit@0.6.125-alpha.d0d38c.0<br/>@codecademy/gamut-labs@31.11.14-alpha.d0d38c.0<br/>@codecademy/gamut@47.3.1-alpha.d0d38c.0<br/>@codecademy/monaco@0.6.44-alpha.d0d38c.0<br/>@codecademy/styleguide@57.2.1-alpha.d0d38c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d5915e8f0b4085e6e2480--gamut-preview.netlify.app>https://627d5915e8f0b4085e6e2480--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d5915e8f0b4085e6e2480>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.1-alpha.fc9ebc.0<br/>@codecademy/gamut-kit@0.6.125-alpha.fc9ebc.0<br/>@codecademy/gamut-labs@31.11.14-alpha.fc9ebc.0<br/>@codecademy/gamut@47.3.1-alpha.fc9ebc.0<br/>@codecademy/monaco@0.6.44-alpha.fc9ebc.0<br/>@codecademy/styleguide@57.2.1-alpha.fc9ebc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d65f1763053035fd8a44a--gamut-preview.netlify.app>https://627d65f1763053035fd8a44a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d65f1763053035fd8a44a>Deploy Logs</a></p>

File: jest.config.base.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const path = require('path');
 
 module.exports = (packageName, overrides) => ({
+  testEnvironment: 'jest-environment-jsdom',
```
Comment (dreamwasp) [inline jest.config.base.js:4]: `node` is now the default `testEnviroment` in the newest jest version

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d6a58c4b9aa083b1652f4--gamut-preview.netlify.app>https://627d6a58c4b9aa083b1652f4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d6a58c4b9aa083b1652f4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.125-alpha.ae868f.0<br/>@codecademy/gamut-labs@31.11.14-alpha.ae868f.0<br/>@codecademy/gamut@47.3.1-alpha.ae868f.0<br/>@codecademy/monaco@0.6.44-alpha.ae868f.0<br/>@codecademy/styleguide@57.2.1-alpha.ae868f.0<br/></pre>

File: packages/gamut/src/GridForm/GridFormInputGroup/GridFormTextInput/index.tsx [hunk 2]
```diff
@@ -37,6 +40,11 @@ export const GridFormTextInput: React.FC<GridFormTextInputProps> = ({
         field?.onUpdate?.(event.target.value);
         await onChange(event);
       }}
+      onFocus={() => {
+        if (error) {
+          clearErrors(field.name);
+        }
+      }}
```
Comment (dreamwasp) [inline packages/gamut/src/GridForm/GridFormInputGroup/GridFormTextInput/index.tsx:47]: this is the actual bug fix

File: packages/gamut/src/GridForm/GridFormInputGroup/GridFormTextArea/index.tsx [hunk 3]
```diff
@@ -37,6 +40,11 @@ export const GridFormTextArea: React.FC<GridFormTextAreaProps> = ({
         field?.onUpdate?.(event.target.value);
         await onChange(event);
       }}
+      onFocus={() => {
+        if (error) {
+          clearErrors(field.name);
+        }
+      }}
```
Comment (dreamwasp) [inline packages/gamut/src/GridForm/GridFormInputGroup/GridFormTextArea/index.tsx:47]: this is the actual bug fix, again

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.125-alpha.aed946.0<br/>@codecademy/gamut-labs@31.11.14-alpha.aed946.0<br/>@codecademy/gamut@47.3.1-alpha.aed946.0<br/>@codecademy/monaco@0.6.44-alpha.aed946.0<br/>@codecademy/styleguide@57.2.1-alpha.aed946.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d736d4b138d0d04c51d5d--gamut-preview.netlify.app>https://627d736d4b138d0d04c51d5d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d736d4b138d0d04c51d5d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.125-alpha.3ecf5b.0<br/>@codecademy/gamut-labs@31.11.14-alpha.3ecf5b.0<br/>@codecademy/gamut@47.3.1-alpha.3ecf5b.0<br/>@codecademy/monaco@0.6.44-alpha.3ecf5b.0<br/>@codecademy/styleguide@57.2.1-alpha.3ecf5b.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.126-alpha.e6dcc5.0<br/>@codecademy/gamut-labs@31.12.1-alpha.e6dcc5.0<br/>@codecademy/gamut@47.3.1-alpha.e6dcc5.0<br/>@codecademy/monaco@0.6.44-alpha.e6dcc5.0<br/>@codecademy/styleguide@57.3.1-alpha.e6dcc5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627ec2540f371e1310efcded--gamut-preview.netlify.app>https://627ec2540f371e1310efcded--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627ec2540f371e1310efcded>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.127-alpha.099c3f.0<br/>@codecademy/gamut-labs@31.12.2-alpha.099c3f.0<br/>@codecademy/gamut@47.3.2-alpha.099c3f.0<br/>@codecademy/monaco@0.6.45-alpha.099c3f.0<br/>@codecademy/styleguide@57.4.1-alpha.099c3f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6282771a52a20f45eef30519--gamut-preview.netlify.app>https://6282771a52a20f45eef30519--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6282771a52a20f45eef30519>Deploy Logs</a></p>

File: jest.config.base.js (hunk 1 above)
Comment (jakemhiller) [reply jest.config.base.js:4]: 👍

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6283fec8cd5e590d688dcd47--gamut-preview.netlify.app>https://6283fec8cd5e590d688dcd47--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6283fec8cd5e590d688dcd47>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.128-alpha.4bb059.0<br/>@codecademy/gamut-labs@31.12.3-alpha.4bb059.0<br/>@codecademy/gamut@47.3.3-alpha.4bb059.0<br/>@codecademy/monaco@0.6.46-alpha.4bb059.0<br/>@codecademy/styleguide@57.4.2-alpha.4bb059.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://628514e14a6fbe32f4645eac--gamut-preview.netlify.app>https://628514e14a6fbe32f4645eac--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/628514e14a6fbe32f4645eac>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6298d38e15b11738c554f1d9--gamut-preview.netlify.app>https://6298d38e15b11738c554f1d9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6298d38e15b11738c554f1d9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.135-alpha.ba4ee4.0<br/>@codecademy/gamut-labs@31.15.3-alpha.ba4ee4.0<br/>@codecademy/gamut@47.3.4-alpha.ba4ee4.0<br/>@codecademy/monaco@0.6.47-alpha.ba4ee4.0<br/>@codecademy/styleguide@57.7.3-alpha.ba4ee4.0<br/></pre>

**Review by aresnik11 — APPROVED** [review]
Love the consolidation of 6 basically identical tests into one test that runs on all sub components. All tests pass so LGTM

**Review by thedoomshine — APPROVED** [review]
LGTM 👍🏻 🔥 ✨

**Pull request merged**